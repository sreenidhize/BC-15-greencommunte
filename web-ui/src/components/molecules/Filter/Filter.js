import React from "react";
import PropTypes from 'prop-types';
import {makeStyles, FormControlLabel} from "@material-ui/core";
import FilterCheckbox from "../../atoms/filterCheckbox/FilterCheckbox";

const useStyles = makeStyles((theme) => ({
    checkboxLabel: {
        fontFamily: theme.typography.fontFamily,
        fontSize: '14px',
        letterSpacing: '0.1px',
        width: '74px',
        height: '22px',
        lineHeight: '1.57',
        color: '#5f7381'
    }
}));

const Filter = ({label, filters, handleChange}) => {
    const booleanValue = (checkboxFilters, checkboxLabel) => {
        for(let i = 0; i < checkboxFilters.length; i++) {
            if(checkboxFilters[i].label === checkboxLabel) {
                return checkboxFilters[i].checked;
            }
        }
    };
    const classes = useStyles();
    const [select, setSelect] = React.useState(booleanValue(filters, label));
    const afterSelected = () => {
        setSelect(!select);
    };
    const handleCheckBox = (e) => {
        afterSelected();
        handleChange(e.target.checked, label);
    };
    return (
            <FormControlLabel
                data-testid="filter"
                control={<FilterCheckbox checked={select} onChange={handleCheckBox} name={label} />}
                label={<span className={classes.checkboxLabel} >{label}</span>}  
            />
    );
};

Filter.propTypes = {
    label: PropTypes.string.isRequired,
    isChecked: PropTypes.bool,
    handleChange: PropTypes.func,
};

export default Filter;