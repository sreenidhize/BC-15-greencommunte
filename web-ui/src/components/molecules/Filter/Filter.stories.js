import React from 'react';
import Filter from "./Filter";
import {action} from "@storybook/addon-actions";

export default {
    title: 'Molecules/Filter',
    component: Filter
};

const Template = (args) => <Filter {...args} />;

export const CheckBoxFilter = Template.bind({});
CheckBoxFilter.args = {
    label: '0-10 km',
    handleChange: action(" Filter checkbox clicked")
};