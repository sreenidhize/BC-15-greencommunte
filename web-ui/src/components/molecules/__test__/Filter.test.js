import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Filter from "../Filter/Filter";

describe("Filter", () => {
    it("Matches the snapshot", () => {
        const { container } = render(
            <Filter label={"Internship"} filters={[]} handleChange={()=>{}}/>
        );
        expect(container).toMatchSnapshot();
    });

    test("test filter render", () => {
        const { getByTestId } = render(<Filter label={"Internship"} filters={[]} handleChange={()=>{}}/>);
        expect(getByTestId("filter")).toBeInTheDocument();
    });
});