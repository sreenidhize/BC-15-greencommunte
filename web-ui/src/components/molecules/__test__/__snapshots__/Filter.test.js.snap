// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Matches the snapshot 1`] = `
<div>
  <label
    class="MuiFormControlLabel-root"
    data-testid="filter"
  >
    <span
      aria-disabled="false"
      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiCheckbox-root Component-root-2"
      data-testid="filter-checkbox"
    >
      <span
        class="MuiIconButton-label"
      >
        <input
          class="PrivateSwitchBase-input-7"
          data-indeterminate="false"
          name="Internship"
          type="checkbox"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </span>
    <span
      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
    >
      <span
        class="makeStyles-checkboxLabel-1"
      >
        Internship
      </span>
    </span>
  </label>
</div>
`;
